src/utils/convert.ts

```diff
@@ -99,7 +99,8 @@ export function parsePartitionBytes32Value(value: string): Partition {
 
 export function parseBallotStageValue(value: BigNumber): BallotStage {
   let stageResult: BallotStage = BallotStage.Prep;
-  switch (value.toNumber()) {
+  const numStage = typeof value === 'number' ? value : value.toNumber();
```

Comment:
Why is this ternary necessary?

Reply:
0x takes zero values and convert it to number in a low level logic. To be sure and handle that, I decided to did this ternary.

Reply:
Please leave a comment explaining this (hopefully with a link to the relevant github issue or part of the 0x documentation that talks about this)

Reply:
https://github.com/0xProject/0x-monorepo/blob/development/packages/utils/src/abi_encoder/evm_data_types/int.ts

`if (this._width === constants.NUMBER_OF_BYTES_IN_INT8) {
   return value.toNumber();
}`